src/Utils/QueryAnalyzer.php

```diff
@@ -531,16 +556,95 @@ public function get_headers( array $headers = [] ): array {
 				if ( 0 === $index ) {
-					$headers['X-GraphQL-Keys']      = $chunk;
-					$headers['X-GraphQL-Keys-Size'] = strlen( $chunk );
+					$return_keys = $chunk;
 				} else {
-					$headers[ 'X-GraphQL-Skipped-Keys-' . $index ]           = $chunk;
-					$headers[ 'X-GraphQL-Skipped-Keys-' . $index . '-Size' ] = strlen( $chunk );
+					$this->skipped_keys = trim( $this->skipped_keys . ' ' . $chunk );
 				}
 			}
 		}
 
+		$skipped_keys_array = ! empty( $this->skipped_keys ) ? explode( ' ', $this->skipped_keys ) : '';
+
+		$this->graphql_keys = [
+			'keys'             => $return_keys,
+			'keysLength'       => strlen( $return_keys ),
+			'skippedKeys'      => $this->skipped_keys,
+			'skippedKeysCount' => ! empty( $skipped_keys_array ) ? count( $skipped_keys_array ) : 0,
+		];
+
+		return $this->graphql_keys;
+
+	}
+
+
+	/**
+	 * Return headers
+	 *
+	 * @param array $headers The array of headers being returned
+	 *
+	 * @return array
+	 */
+	public function get_headers( array $headers = [] ): array {
+
+		$keys = $this->get_graphql_keys();
+
+		if ( ! empty( $keys ) ) {
+
+			$headers['X-GraphQL-Query-ID']  = $this->query_id ?: null;
+			$headers['X-GraphQL-Keys']      = $keys['keys'] ?: null;
+			$headers['X-GraphQL-Keys-Size'] = strlen( $keys['keys'] ) ?: 0;
+			if ( 0 < absint( $keys['skippedKeysCount'] ) ) {
+				$headers['X-GraphQL-Skipped-Keys-Count'] = $keys['skippedKeysCount'] ?: 0;
+			}       
+		}
+
 		return $headers;
 	}
 
+	/**
+	 *
+	 * @param mixed    $response
+	 * @param WPSchema $schema         The WPGraphQL Schema
+	 * @param string   $operation_name The operation name being executed
+	 * @param string   $request        The GraphQL Request being made
+	 * @param array    $variables      The variables sent with the request
+	 *
+	 * @return array|mixed
+	 */
+	public function show_query_analyzer_in_extensions( $response, $schema, $operation_name, $request, $variables ) {
+
+
+		$should = \WPGraphQL::debug();
+
+		/**
+		 * @param bool     $should         Whether the query analyzer output should be displayed in the Extensions output. Default to the value of WPGraphQL Debug.
+		 * @param mixed    $response       The response of the WPGraphQL Request being executed
+		 * @param WPSchema $schema         The WPGraphQL Schema
+		 * @param string   $operation_name The operation name being executed
+		 * @param string   $request        The GraphQL Request being made
+		 * @param array    $variables      The variables sent with the request
+		 */
+		$should_show_query_analyzer_in_extensions = apply_filters( 'graphql_should_show_query_analyzer_in_extensions', $should, $response, $schema, $operation_name, $request, $variables );
+
+		// If the query analyzer output is disabled,
+		// don't show the output in the response
+		if ( false === $should_show_query_analyzer_in_extensions ) {
+			return $response;
+		}
+
+		$keys = $this->get_graphql_keys();
+
+		if ( ! empty( $response ) ) {
+			if ( is_array( $response ) ) {
```

Comment:
Been looking at [the `graphql_request_results` filter](https://github.com/jasonbahl/wp-graphql/blob/522dcbcbfc4d540912972ebd87640c65b9c594c5/src/Request.php#L563). I'm not entire clear under which circumstances a `$request` is an `object` and not an `array`, but we should probably update the docblocks there that currently only allow for an `array` type.

(doesnt have to happen in _this_ pr, but we should at least update the `@return` here to `array|object|null`).